File: 2015/day02/day02.py
def dims(line):
    return sorted(map(int, line.split('x')))

with open('day02.input') as file:
    print(sum(3*l*w + 2*h*w + 2*h*l for l, w, h in map(dims, file)))

    file.seek(0)

    print(sum(2*l + 2*w + l*w*h for l, w, h in map(dims, file)))

File: 2015/day02/day02.nim
import algorithm, math, sequtils, strutils

var a = "day02.input".readFile
  .splitLines[0..^2].mapIt(it.split('x').map(parseInt).sorted(cmp))

echo a.mapIt(3*it[0]*it[1] + 2*it[1]*it[2] + 2*it[0]*it[2]).sum()
echo a.mapIt(2*it[0] + 2*it[1] + it[0]*it[1]*it[2]).sum()
